packages/cli/src/generators/app.ts

```diff
@@ -37,6 +40,19 @@ class AppGenerator extends Generator<AppGeneratorOptions> {
   }
 
   async postWrite() {
+    // Overwrite dependency versions just before installing
+    const pkg = readJSONSync(join(this.destinationPath(), 'package.json'))
+    const pkgDependencies = Object.keys(pkg.dependencies)
+    const latestVersions = await Promise.all(
+      pkgDependencies.map(async (dependency) => {
+        const res = await fetch(`https://registry.npmjs.org/-/package/${dependency}/dist-tags`)
+        const json = await res.json()
+        return json.latest as string
+      }),
+    )
+    pkg.dependencies = pkgDependencies.reduce((o, k, i) => ({...o, [k]: latestVersions[i]}), {})
+    writeJSONSync(join(this.destinationPath(), 'package.json'), pkg)
```

Comment:
What are the ways this could fail and how can we handle it well?

We should probably add 1 or 2 retries here. And maybe then fallback to whatever the template originally had for the failing dependency?

Reply:
I now have it retry a few times, and use the old versions if any fetching fails.